### ISSUE TITLE
Transplanting TreeSelect, Transfer, Upload from antd?

### ISSUE THREAD

**Issue by WisdomFusion** [NG-ZORRO/ng-zorro-antd#132]
<pre><code>
[ ] Bug report
[x] Feature request
[ ] Documentation issue or request
[ ] Regression (a behavior that used to work and stopped working in a new release)
[ ] Support request => Please do not submit support request here
</code></pre>

## Current behavior

We can't find TreeSelect, Transfer, Upload components till now.

## Expected behavior

Is there a schedule for transplanting those components to ng-zorro from antd?

Looking forward for the whole set of components from antd.

Thanks.

**Comment by vthinkxie** [comment]
The most important thing we need to do is to keep all bugs clean until `0.5.0` final version released.
TreeSelect Transfer and Upload components is on the schedule, but we can not promise the exact time.

**Comment by thegatheringstorm** [comment]
Can I claim the Upload components?
@vthinkxie

**Comment by thegatheringstorm** [comment]
Upload component has been mentioned in #89.
@vthinkxie

**Comment by thegatheringstorm** [comment]
Should the basic Upload Component be abstracted as described in react-component/upload and https://github.com/ant-design/ant-design/tree/master/components/upload ?

**Comment by vthinkxie** [comment]
@thegatheringstorm 
Just keep the api and style https://github.com/ant-design/ant-design/tree/master/components/upload/style same as react version

**Comment by thegatheringstorm** [comment]
Okay

**Comment by thegatheringstorm** [comment]
Should I send you all the code for the upload component at once?Or,Send you the basic code so that you can judge whether my architecture has any problems earlier.  @vthinkxie

**Comment by vthinkxie** [comment]
@thegatheringstorm 
You can submit your pr any time, we will review it after you finish your work.
thanks a lot.

**Comment by thegatheringstorm** [comment]
Okay  😄

**Comment by thegatheringstorm** [comment]
HttpModule is only needed when uploading.
Should I
```js
import { HttpModule }    from '@angular/http';
```
in AppComponent?
I really don't know how to deal with this.

And,in react-component/upload,a new tool has been written there.
[Request.js](https://github.com/react-component/upload/blob/master/src/request.js)

Which should I use (Request.js or the Http in Angular)?
> The essence of the question is whether I should maintain consistency of style or make full use of the Angular

**Comment by vthinkxie** [comment]
@thegatheringstorm please use the HttpClientModule
 `import { HttpClientModule } from '@angular/common/http'`
you can find more information here
https://angular.cn/guide/http

**Comment by thegatheringstorm** [comment]
如何在 **_组件_** 中使用 HttpClientModule ？能仅仅在组件中使用 HttpClientModule 而不必把该组件转换为 Module 吗？
```js
// app.module.ts:

import {NgModule} from '@angular/core';
import {BrowserModule} from '@angular/platform-browser';

// Import HttpClientModule from @angular/common/http
import {HttpClientModule} from '@angular/common/http';

@NgModule({
  imports: [
    BrowserModule,
    // Include it under 'imports' in your application module
    // after BrowserModule.
    HttpClientModule,
  ],
})
export class MyAppModule {}
```
这里只有定义为 MyAppModule(Module) 才能使用 HttpClientModule 吧。可是我们的上传组件是一个组件不是 Module 啊。
还望指教 :smiley:

**Comment by vthinkxie** [comment]
@thegatheringstorm have a look at the source code, all `components` are export as modules

**Comment by thegatheringstorm** [comment]
Thanks

**Comment by murraybauer** [comment]
Looks like Tree comp has been added now, but still missing Tree Select.

**Comment by wanghaolu** [comment]
@vthinkxie treeselect组件有计划开发么，这个在复杂表单中应用还是挺多的

**Comment by vthinkxie** [comment]
treeselect将在1.0正式版中放出

**Comment by vthinkxie** [comment]
closed by #1477
